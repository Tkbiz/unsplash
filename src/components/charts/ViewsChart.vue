<script>
import {Bar} from 'vue-chartjs'

export default {
  extends: Bar,
  props: {
    photoViews: {
      type: Object,
      required: true
    }
  },
  mounted() {
    this.renderChart({
      labels: this.photoViews.historical.values.map(date => date.date),
      datasets: [
        {
          label: 'Photo\'s views',
          data: this.photoViews.historical.values.map(value => value.value),
          fill: false,
          borderColor: '#57a773',
          backgroundColor: '#57a773',
          borderWidth: 1
        }
      ]
    }, {
      scales: {
        yAxes: [{
          ticks: {
            beginAtZero: true
          },
          gridLines: {
            display: true
          }
        }],
        xAxes: [{
          gridLines: {
            display: false
          }
        }]
      },
      legend: {
        display: true
      },
      responsive: true,
      maintainAspectRatio: false
    })
  }
}
</script>
