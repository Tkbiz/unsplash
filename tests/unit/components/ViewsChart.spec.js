import {mount, createLocalVue} from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import ViewsChart from '@/components/charts/ViewsChart'
import {Bar} from 'vue-chartjs'

Vue.use(Vuetify)

const localVue = createLocalVue()

jest.mock('vue-chartjs')

describe('ViewsChart.vue', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    wrapper = mount(ViewsChart, {
      localVue,
      vuetify,
      propsData: {
        photoViews: {
          historical: {
            values: []
          }
        }
      }
    })
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
