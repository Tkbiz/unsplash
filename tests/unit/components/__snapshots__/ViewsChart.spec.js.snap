// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewsChart.vue should match snapshot 1`] = `""`;
